tool/src/callback_dictionary.ts

```diff
@@ -1,28 +1,130 @@
 import hljs from "highlight.js";
 import mathematica from "highlight.js/lib/languages/mathematica";
+import javascript from "highlight.js/lib/languages/javascript";
+import typescript from "highlight.js/lib/languages/typescript";
+import php from "highlight.js/lib/languages/php";
+import ruby from "highlight.js/lib/languages/ruby";
 
 import { expand, StringObject } from "./util.js"
+import _ from "lodash";
 
-export const callbackDictionary = new Map<string, string>([
-  [
-    hljs.END_SAME_AS_BEGIN({})["on:begin"]!.toString(),
-    "endSameAsBeginOnBegin"
-  ],
-  [
-    hljs.END_SAME_AS_BEGIN({})["on:end"]!.toString(),
-    "endSameAsBeginOnEnd"
-  ],
+const commonCallbacks = new Map<string, string>(
   [
-    hljs.SHEBANG()["on:begin"]!.toString(),
-    "shebangOnBegin",
-  ],
+    [
+      hljs.END_SAME_AS_BEGIN({})["on:begin"]!.toString(),
+      "endSameAsBeginOnBegin"
+    ],
+    [
+      hljs.END_SAME_AS_BEGIN({})["on:end"]!.toString(),
+      "endSameAsBeginOnEnd"
+    ],
+    [
+      hljs.SHEBANG()["on:begin"]!.toString(),
+      "shebangOnBegin",
+    ],
+  ]
+);
+
+export const callbackDictionary = new Map<string, string>([
+  ...commonCallbacks,
 
   ...expand([
-    mathematica,
+    javascript(hljs),
+    mathematica(hljs),
+    php(hljs),
+    ruby(hljs),
+    typescript(hljs),
   ].map(language => getLanguageCallbacks(language))),
 ]);
 
 function getLanguageCallbacks(language: StringObject<any>): [string, string][] {
-  // TODO: Extract all callbacks recursively.
-  return [["a", "b"]];
+  let result: [string, string][] = [];
+
+  const addResult = (k: string, v: string) => {
+    if (!commonCallbacks.has(v.toString())) {
```

Comment:
Add the language name to the key.

Reply:
The problem here is that the key is the source code of the callback that we match when replacing it with the function name.

I couldn't find a way how we can casually add language name to the key.
My suggestion is to leave everything as is, and if there are new callbacks in `highlight.js`, we would need to make manual changes anyway. So, I think it is not that big of a deal.